src/components/CurrencyConverter.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import { Form, Field } from 'react-final-form';
+import { H2 } from './styledElements/H2';
+import { Button } from './styledElements/Button';
+import { Loader } from './styledElements/Loader';
+import { FormStyles } from './styledElements/FormStyles';
+import { API_KEY, API_URL } from '../utils/apiConstants';
+
+const initialValues = {
+  currFrom: 'PLN',
+  amount: 0,
+  currTo: 'EUR'
+};
+
+const validate = values => {
+  const errors = {};
+
+  if (values.currFrom === values.currTo) {
+    errors.currFrom = 'Wybrałeś dwie takie same waluty.';
+  }
+  if (values.amount <= 0) {
+    errors.amount = 'Podaj kwotę większą od 0.';
+  }
+  const amountIsNum = (new RegExp('[0-9]')).test(values.amount);
+  if (!amountIsNum) {
+    errors.amount = 'Wpisz liczbę.';
+  }
+
+  return errors;
+};
+
+export const CurrencyConverter = props => {
+  const history = useHistory();
+  const { currenciesList } = props;
+
+  const onSubmit = async values => {
+    const { currFrom, currTo, amount } = values;
+    const query = `${currFrom}_${currTo}`;
+    const response = await fetch(
+      `${API_URL}/api/v7/convert?q=${query}&compact=ultra&apiKey=${API_KEY}`
+    );
+    const data = await response.json();
+
+    const result = data[query] * Number(amount);
+    const fixedResult = result.toFixed(2);
+
+    props.addToHistory({
+      currFrom,
+      currTo,
+      amountFrom: amount,
+      result: fixedResult,
+      date: new Date().getTime()
+    });
+
+    history.push('/history');
+  };
+
+  return (
+    <>
+      <H2>Przelicznik walut</H2>
+
+      {!currenciesList.length && <Loader />}
+
+      {!!currenciesList.length && (
+        <Form
```

Comment:
Warto użyć subscription, co pozwoli na unieknięcie zbędnym renderów pól formularza.

Reply:
Próbowałam tutaj użyć subscription zgodnie z tym jak jest to wskazane w dokumentacji, ale pojawił się wówczas problem z dostępem do obiektu _values_ wewnątrz komponentu formularza.